examples/example-server-lib/floating_window_manager.cpp

```diff
@@ -605,3 +617,159 @@ void FloatingWindowManagerPolicy::handle_modify_window(WindowInfo& window_info,
 
     MinimalWindowManager::handle_modify_window(window_info, mods);
 }
+
+void FloatingWindowManagerPolicy::try_place_new_window_and_account_for_occlusion(WindowSpecification& parameters)
+{
+    namespace geometry = mir::geometry;
+    if (parameters.state() == mir_window_state_restored)
+    {
+        std::vector<geometry::Rectangle> window_rects;
+        tools.for_each_window_in_workspace(
+            active_workspace,
+            [this, &window_rects](Window const& window)
+            {
+                auto const& info = tools.info_for(window);
+
+                // Skip invisible windows
+                if (!info.is_visible() || info.depth_layer() != mir_depth_layer_application)
+                    return;
+
+                window_rects.emplace_back(window.top_left(), window.size());
+            }
+        );
+
+        // Worst case scenario: you try taking a hole out of the middle of
+        // a rect, results in 4 rectangles. Every other case is a special
+        // case of this one where the width or height of the resulting
+        // rects are < 0.
+        auto const subtract_rectangles = [](geometry::Rectangle rect,
+                                            geometry::Rectangle hole) -> std::vector<geometry::Rectangle>
```

Comment:
```suggestion
        auto const subtract_rectangles = [](geometry::Rectangle first,
                                            geometry::Rectangle second) -> std::vector<geometry::Rectangle>
```

Is the name `hole` a poor one here?

Reply:
It made more sense at the time, you're subtracting a _hole_ from a _rect_. Of course, this naming only makes sense in the worst case scenario (the hole is wholly contained within the rect), but all other cases can be derived from this base case.